airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultCheckConnectionWorker.java

```diff
@@ -84,6 +89,26 @@ public ConnectorJobOutput run(final StandardCheckConnectionInput input, final Pa
           .map(AirbyteMessage::getConnectionStatus)
           .findFirst();
 
+      if (input.getActorId() != null && input.getActorType() != null) {
+        LOGGER.info("has actor and type");
+        final Optional<AirbyteControlConnectorConfigMessage> optionalConfigMsg = WorkerUtils.getLastConfigControlMessage(messagesByType);
```

Comment:
It looks like `messagesByType` will store _all_ the AirbyteMessages we get from the container, arranged by type.  Not that you added this in this PR, but that _could_ be huge.  I guess since this is the CHECK method, it's not likely to grow that big...